src/core/metrics/sources/nginx_access_log.go

```diff
@@ -436,6 +475,56 @@ func getAverageMetricValue(metricValues []float64) float64 {
 	return value
 }
 
+func (c *NginxAccessLog) getHttpStatus(status string, counter map[string]float64) {
+
+	if v, err := strconv.Atoi(status); err == nil {
+		n := fmt.Sprintf("status.%dxx", v/100)
+		counter[n] = counter[n] + 1
+		switch v {
+		case 403, 404, 500, 502, 503, 504:
+			n := fmt.Sprintf("status.%d", v)
+			counter[n] = counter[n] + 1
+		case 499:
+			n := "status.discarded"
+			counter[n] = counter[n] + 1
+		case 400:
+			n := "request.malformed"
+			counter[n] = counter[n] + 1
+		}
+	} else {
+		c.logger.Log(fmt.Sprintf("Error getting status value from access logs, %v", err))
+	}
+}
+
+func getUpstreamCacheStatus(status string, counter map[string]float64) {
+
+	n := fmt.Sprintf("cache.%s", strings.ToLower(status))
+
+	switch status {
+	case "BYPASS":
+		counter[n] = counter[n] + 1
+		return
+	case "EXPIRED":
+		counter[n] = counter[n] + 1
+		return
+	case "HIT":
+		counter[n] = counter[n] + 1
+		return
+	case "MISS":
+		counter[n] = counter[n] + 1
+		return
+	case "REVALIDATED":
+		counter[n] = counter[n] + 1
+		return
+	case "STALE":
+		counter[n] = counter[n] + 1
+		return
+	case "UPDATING":
+		counter[n] = counter[n] + 1
+		return
+	}
```

Comment:
nit: could condense a little

```suggestion
	switch status {
	case "BYPASS", "EXPIRED", "HIT", "MISS", "REVALIDATED", "STALE", "UPDATING":
		counter[n] = counter[n] + 1
	default:
		// ignore unknown status	
	}
```